dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/manager/DefaultExecutorRepository.java

```diff
@@ -177,8 +186,41 @@ public ExecutorService nextExecutorExecutor() {
     }
 
     @Override
-    public ScheduledExecutorService getServiceExporterExecutor() {
-        return serviceExporterExecutor;
+    public ScheduledExecutorService getExportReferExecutor() {
+        if (exportReferExecutor == null) {
+            synchronized (LOCK) {
+                if (exportReferExecutor == null) {
+                    int coreSize = getExportReferThreadNum();
+                    exportReferExecutor = Executors.newScheduledThreadPool(coreSize,
+                        new NamedThreadFactory("Dubbo-export-refer", true));
+                }
+            }
+        }
+
+        return exportReferExecutor;
+    }
+
+    private Integer getExportReferThreadNum() {
+        Stream<Integer> provider = getConfigManager().getProviders()
```

Comment:
This might cause NPE in some situations. It is better to return default value.

Reply:
I don't think this is necessary, there always exists default provider/consumer.